src/test/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutesV2Spec.scala

```diff
@@ -154,4 +154,41 @@ class UserRoutesV2Spec extends AnyFlatSpec with Matchers with ScalatestRouteTest
       responseAs[SamUserAllowances] should be(SamUserAllowances(allowed = false, enabled = false, termsOfService = false))
     }
   }
+
+  "GET /api/users/v2/self/attributes" should "get the user attributes of the calling user" in {
+    // Arrange
+    val userAttributes = SamUserAttributes(defaultUser.id, marketingConsent = true)
+
+    val samRoutes = new MockSamRoutesBuilder(allUsersGroup)
+      .withEnabledUser(defaultUser) // "persisted/enabled" user we will check the status of
+      .withAllowedUser(defaultUser) // "allowed" user we will check the status of
+      .withUserAttributes(defaultUser, userAttributes)
+      .callAsNonAdminUser()
+      .build
+
+    // Act and Assert
+    Get(s"/api/users/v2/self/attributes") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.OK
+      responseAs[SamUserAttributes] should be(userAttributes)
+    }
+  }
+
+  "PATCH /api/users/v2/self/attributes" should "update the user attributes of the calling user" in {
+    // Arrange
+    val userAttributesRequest = SamUserAttributesRequest(marketingConsent = Some(false))
+    val userAttributes = SamUserAttributes(defaultUser.id, marketingConsent = false)
+
+    val samRoutes = new MockSamRoutesBuilder(allUsersGroup)
+      .withEnabledUser(defaultUser) // "persisted/enabled" user we will check the status of
+      .withAllowedUser(defaultUser) // "allowed" user we will check the status of
+      .withUserAttributes(defaultUser, userAttributes)
+      .callAsNonAdminUser()
+      .build
+
+    // Act and Assert
+    Patch(s"/api/users/v2/self/attributes", userAttributesRequest) ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.OK
+      responseAs[SamUserAttributes] should be(userAttributes)
+    }
+  }
```

Comment:
Should we have one where requesting another user's attributes is denied or is that already covered?

Reply:
That's handled in `CreateUserSpec`. Since this is an API layer test, its really only testing to make sure that the ser/der works, and probably shouldn't be testing business logic.

Reply:
I think in the case of a `Not Found` we should be handling that n the api layer since currently that is where the not found logic is. I think in an ideal world we would have some sort of adapter between the routes and the services which would map all the objects returned from the services layer into their corresponding response codes, but currently that logic is scattered between the service and routes layers